src/gameLogic/execution/gadget/Cocktail.cpp

```diff
@@ -79,5 +101,26 @@ namespace spy::gameplay {
 
         return ret;
     }
+
+    bool GadgetExecutor::executeCocktail(State &s, const GadgetAction &a, const MatchConfig &config) {
+        auto sourceChar = s.getCharacters().getByUUID(a.getCharacterId());
+        auto &targetField = s.getMap().getField(a.getTarget());
+        bool drink = (sourceChar->getCoordinates().value() == a.getTarget());
+
+        if (drink) {
+            // drink the cocktail
+            return executeDrink(s, config, sourceChar);
+        } else if (targetField.getFieldState() == scenario::FieldStateEnum::BAR_TABLE
+                   && targetField.getGadget().has_value()) {
```

Comment:
```suggestion
        } else if (targetField.getFieldState() == scenario::FieldStateEnum::BAR_TABLE {
```
Überprüfung "unnötig", da Voraussetzung für Execute ist, dass Aktion valide ist.